File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int MAX_W=0, MAX_H=0;
    for (auto size: sizes) {
        if(size[0] < size[1]) swap(size[0],size[1]);
        MAX_W = max(size[0], MAX_W);
        MAX_H = max(size[1], MAX_H);
    }
    answer = MAX_W * MAX_H;
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let [max_w, max_h] = [0, 0]
    
    for (let i = 0; i < sizes.length; i++) {
        let [w, h] = sizes[i]
        
        if (w < h) {
            const temp = w
            w = h
            h = temp
        }
        
        max_w = Math.max(max_w, w)
        max_h = Math.max(max_h, h)
    }
    return max_w * max_h
}